### ISSUE TITLE
stac-to-dc doesnt support multipolygon

### ISSUE THREAD

**Issue by pindge** [opendatacube/odc-tools#538]
```
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/odc/apps/dc_tools/stac_api_to_dc.py", line 194, in stac_api_to_odc
    _ = future.result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 437, in result
    return self.__get_result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
    raise self._exception
  File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.8/dist-packages/odc/apps/dc_tools/stac_api_to_dc.py", line 132, in process_item
    index_update_dataset(
  File "/usr/local/lib/python3.8/dist-packages/odc/apps/dc_tools/utils.py", line 214, in index_update_dataset
    raise IndexingException(
odc.apps.dc_tools.utils.IndexingException: Exception thrown when trying to create dataset: 'Invalid values passed to Polygon constructor'
 The URI was https://planetarycomputer.microsoft.com/api/stac/v1/collections/io-lulc/items/01K-2020
```

**Comment by omad** [comment]
I'm not even sure that ODC supports Datasets with a Multipolygon extent...

**Comment by alexgleith** [comment]
I think I've hit this same issue indexing this document: https://planetarycomputer.microsoft.com/api/stac/v1/collections/io-lulc-9-class/items/29V-2021

Error looks like this:
```
03/01/2023 01:06:34: ERROR: Failed to handle item 29V-2021 with exception setting an array element with a sequence. The requested array has an inhomogeneous shape after 2 dimensions. The detected shape was (5, 1) + inhomogeneous part.
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/odc/apps/dc_tools/stac_api_to_dc.py", line 167, in stac_api_to_odc
    _ = future.result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 437, in result
    return self.__get_result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
    raise self._exception
  File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.8/dist-packages/odc/apps/dc_tools/stac_api_to_dc.py", line 112, in process_item
    meta, uri = item_to_meta_uri(item, rewrite, rename_product)
  File "/usr/local/lib/python3.8/dist-packages/odc/apps/dc_tools/stac_api_to_dc.py", line 98, in item_to_meta_uri
    metadata = stac_transform_absolute(metadata)
  File "/usr/local/lib/python3.8/dist-packages/odc/apps/dc_tools/_stac.py", line 263, in stac_transform_absolute
    return stac_transform(input_stac, relative=False)
  File "/usr/local/lib/python3.8/dist-packages/odc/apps/dc_tools/_stac.py", line 397, in stac_transform
    stac_odc["geometry"] = transform_geom_json_coordinates_to_list(geometry.json)
  File "/usr/local/lib/python3.8/dist-packages/odc/apps/dc_tools/_stac.py", line 407, in transform_geom_json_coordinates_to_list
    geom_json["coordinates"] = numpy.array(
ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 2 dimensions. The detected shape was (5, 1) + inhomogeneous part.
```

**Comment by omad** [comment]
Fixed by #554
